### PR TITLE
Add a new package for MCB

### PR DESCRIPTION
**Pull request by akhilpatlolla**

### CODE REVIEW

File: var/spack/repos/builtin/packages/mcb/package.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    build_directory = 'src'
+
+    @property
+    def build_targets(self):
+        targets = []
+        targets.append(
+            'BOOST_INCLUDE = -I{0} '.
+            format(self.spec['boost'].prefix.include.boost))
+        targets.append(
+            'BOOST_LIB = -L{0} -Wl,-rpath {0}'.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mcb/package.py]: This `-Wl,-rpath` syntax won't work for every compiler. Notably, NAG uses a different format. Either use `self.compiler.cc_rpath_args` (replace `cc` with the correct compiler) or let Spack's compiler wrappers handle RPATHing.

File: var/spack/repos/builtin/packages/mcb/package.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            format(self.spec['boost'].prefix.lib))
+        targets.append('CXX = %s' % self.spec['mpi'].mpicc)
+        targets.append('OPENMPFLAG = %s' % self.compiler.openmp_flag)
+        targets.append('MPI_INCLUDE = -I%s' % self.spec['mpi'].prefix.include)
+        targets.append('CXXFLAGS = -O2 -xHost $(CXXDEFINES) $(OPENMPFLAG)')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mcb/package.py]: I believe `-xHost` only works for the Intel compilers.

Comment (akhilpatlolla) [inline var/spack/repos/builtin/packages/mcb/package.py]: Tried removing `-xHost`, it's giving a new error 
`
==> Using default implementation: skipping edit phase.
==> 'make' 'BOOST_INCLUDE = -I/home/apatlolla/spack/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/boost-1.63.0-u77poae23raxnqbiguwhwt5adc7dfzg5/include/boost ' 'BOOST_LIB = -L/home/apatlolla/spack/opt/spack/lin$
/home/apatlolla/spack/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/openmpi-2.1.1-bdm67yaoa77o6sioamjnfuilwuiy2hfj/bin/mpicc -O2  -DORTHOXY -DUSE_OPENMP -DUSE_MPI -DNOASSERT  -fopenmp -I. -I.. -I./Meshes -I./Im$
In file included from DomainDecomposition/IMC_Domain_organizerInst.cc:22:0:
DomainDecomposition/IMC_Domain_organizer.cc: In instantiation of ‘void IMC_namespace::IMC_Domain_organizer<mesh_types, mat_types, Source_DB_type>::setUpDomainDecomposition(Source_DB_type&, IMC_namespace:$
DomainDecomposition/IMC_Domain_organizerInst.cc:30:19:   required from here
DomainDecomposition/IMC_Domain_organizer.cc:50:42: error: ‘setUpDomainDecompositionBCs’ was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instant$
     setUpDomainDecompositionBCs( IMC_BCs );
                                          ^
DomainDecomposition/IMC_Domain_organizer.cc:50:42: note: declarations in dependent base ‘IMC_namespace::Domain_organizer<IMC_namespace::Ortho_Cartesian_2D_Mesh, IMC_namespace::photon<unsigned int, Vector$
DomainDecomposition/IMC_Domain_organizer.cc:50:42: note: use ‘this->setUpDomainDecompositionBCs’ instead
make: *** [DomainDecomposition/IMC_Domain_organizerInst.o] Error 1
`

unable to work with internal libraies.

Comment (akhilpatlolla) [inline var/spack/repos/builtin/packages/mcb/package.py]: I  am not sure, 

can I edit the source files, to make those changes in `DomainDecomposition/IMC_Domain_organizer.cc` ?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mcb/package.py]: You can if you want, take a look at `filter_file` and `FileFilter`

File: var/spack/repos/builtin/packages/mcb/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/mcb/package.py]: @akhilpatlolla Can you please take care of the issue raised by @adamjstewart ? Other than that this LGTM.

**Comment by alalazo** [comment]
Closing because no action was taken since September. Feel free to reopen if there are updates.

**Pull request closed without merging**